src/anyio/_core/_subprocess_worker.py

```diff
@@ -0,0 +1,51 @@
+import sys
+from threading import Timer
+
+
+def process_worker():
+    import pickle
+
+    sys.stdout.buffer.write(b'READY\n')
+    while True:
+        idle_timer = Timer(5 * 60, sys.exit)
+        try:
+            command, *args = pickle.load(sys.stdin.buffer)
+        except EOFError:
+            return
+        except BaseException as exc:
+            idle_timer.cancel()
+            exception = exc
+            status = b'EXCEPTION'
+            pickled = pickle.dumps(exc, pickle.HIGHEST_PROTOCOL)
+        else:
+            idle_timer.cancel()
```

Comment:
The process dying in the middle of handling a request causes a `BrokenWorkerProcess` to be raised in the parent.

Reply:
That seems like the wrong response here. Nothing broke in the worker! Yet the user would encounter and have to handle this error as if their processes were flaky.

Reply:
So, should we leave the timing out of processes to the parent? Or implement timeouts some other way on the worker?

Reply:
Actually, since your contract for subprocess lifetime is to limit them to within a trio or asyncio run, I think you could leave timing out of the processes and use a monitoring task to kill them if they have been idle too long.  trio-parallel suffers several extra complications because its workers are intended to span sequential and parallel runs, and so cannot used tasks/nurseries for this purpose.

Reply:
I could use the same task that currently sleeps until cancelled.